src/klipse/klipse_editors.cljs

```diff
@@ -52,7 +52,8 @@
           (swap! state update-in [:eval-counter] inc)
           (let [evaluation-chan (eval-fn (str preamble src-code) @state)
                 first-result (<! evaluation-chan)]
-            (setter first-result)
+              (setter first-result)
```

Comment:
Here, you can handle different cases for `first-result`:

1. if it is a string, behave as before
2. it it is an array, it will be either: `[:ok result]` or `[:err result]`

In `python.cljs`, returns either `[:ok result]` or `[:err result]`.

Dispatch the event from here and not from `python.cljs`

Reply:
Hi @viebel, looks like we are only seeing case # 1:
> if it is a string, behave as before

and never case # 2
do you have any idea why ? We are evaluating python code.

Reply:
@dviramontes you have to modidy the code in `python.cljs` so that it returns
- `[:ok result]` - in case of success
- `[:err error]` - in case of error

In order not to break other languages, you have to handle also cases where a string was returned.
Does it make sense?

Reply:
@viebel yes it does, i'm going to give it a shot and let you know how it goes. Thanks!